packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -332,3 +333,49 @@ export function esbuildCjsExternalPlugin(
     },
   }
 }
+
+// esbuild doesn't transpile `require('foo')` into `import` statements if 'foo' is externalized
+// https://github.com/evanw/esbuild/issues/566#issuecomment-735551834
+export function rollupCjsExternalPlugin(
+  externals: string[],
+  platform: 'node' | 'browser',
+): Rollup.Plugin {
+  const filter = new RegExp(externals.map(matchesEntireLine).join('|'))
+
+  return {
+    name: 'cjs-external',
+    resolveId(id, importer, options) {
+      if (id.startsWith(nonFacadePrefix)) {
+        return {
+          id: id.slice(nonFacadePrefix.length),
+          external: true,
+        }
+      }
+
+      if (filter.test(id)) {
+        // TODO rolldown: kind is not available in rollup
+        const kind: ImportKind = options.custom?.kind
```

Comment:
Rolldown should implemen it, please see https://github.com/rolldown-rs/rolldown/issues/166